include/internal/libfreenect2/depth_packet_processor.h

```diff
@@ -175,6 +175,42 @@ class OpenCLDepthPacketProcessor : public DepthPacketProcessor
 private:
   OpenCLDepthPacketProcessorImpl *impl_;
 };
+
 #endif // LIBFREENECT2_WITH_OPENCL_SUPPORT
+
+class DumpDepthPacketProcessor : public DepthPacketProcessor
+{
+ public:
+  DumpDepthPacketProcessor();
+  virtual ~DumpDepthPacketProcessor();
+
+  virtual void process(const DepthPacket &packet);
+
+  virtual void loadP0TablesFromCommandResponse(unsigned char* buffer, size_t buffer_length);
+  virtual void loadXZTables(const float *xtable, const float *ztable);
+  virtual void loadLookupTable(const short *lut);
+
+  // Get the P0 Tables
+  unsigned char* getP0Tables();
+  size_t getP0TablesLength();
```

Comment:
The length of p0 tables are known, here https://github.com/OpenKinect/libfreenect2/blob/master/include/internal/libfreenect2/protocol/response.h#L220

* `sizeof(P0TablesResponse)`

`loadP0TablesFromCommandResponse()` may get something more than `sizeof(P0TablesResponse)`, but never less.